src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -584,7 +587,13 @@ public DownloadInstance tabularDatafileMetadataAux(@PathParam("fileId") String f
             }
             
             if (auxFile.getIsPublic()) {
-                publiclyAvailable = true;
+                // Require at least one published version for isPublic to be true.
+                for (FileMetadata fm : df.getFileMetadatas()) {
```

Comment:
is it possible to check the if the datafile is published instead? Might be better than going through all the versions? (I can remember if we have a file.isPublished() method, or maybe a file.getPublishicationDate()? and check for !null?

Reply:
Fixed in f55b36862.